velox/exec/HashJoinBridge.cpp

```diff
@@ -64,6 +64,52 @@ void HashJoinBridge::addBuilder() {
   ++numBuilders_;
 }
 
+bool HashJoinBridge::canReclaim() const {
+  return tableSpillFunc_ != nullptr && !probeStarted_.load() &&
+      buildResult_.has_value() && buildResult_->table != nullptr &&
+      buildResult_->table->numDistinct() != 0;
+}
+
+uint64_t HashJoinBridge::reclaim() {
+  VELOX_CHECK(buildResult_.has_value());
+  VELOX_CHECK_NOT_NULL(buildResult_->table);
+
+  auto computeTableReservedBytes = [](std::vector<RowContainer*> allRows) {
+    uint64_t totalReservedBytes{0};
+    for (const auto* rowContainer : allRows) {
+      totalReservedBytes += rowContainer->pool()->reservedBytes();
+    }
+    return totalReservedBytes;
+  };
+  const auto oldMemUsage =
+      computeTableReservedBytes(buildResult_->table->allRows());
+
+  auto spillPartitionSet = tableSpillFunc_(buildResult_->table);
+  buildResult_->table->clear(true);
+
+  const auto reclaimedBytes =
+      oldMemUsage - computeTableReservedBytes(buildResult_->table->allRows());
+
+  auto spillPartitionIdSet = toSpillPartitionIdSet(spillPartitionSet);
```

Comment:
Can you put this code in a function as it is shared with setHashTable?

setHashTableResult
updateHashTableResult 
  -> setSpillPartitionSet?